rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,263 @@
+
+## Pain
+
+- It is not possible to delegate the management of pipelines across a matrix of teams and services.
+- It is not possible to observe individual pipelines (paths on the graph).
+- It is not possible to restrict what pipelines can do (which data they can access, how much data they can send, etc).
+- Managing a large graph is cumbersome and usually results in a very large Vector configuration file that is hard to navigate. Splitting the vector config into multiple files is a creative exercise for the user without any guidance.
+
+## User Experience
+
+- These changes provide a way for Vector users to split their configuration in a way that improves the collaboration between ops and devs.
+- This split will be made by allowing the creation of individual pipeline configuration files, intended to align with services and teams, enabling autonomous management.
+- The ops will now configure their `sources`, `sinks` and `transforms`, and expose them to be used by the devs.
+- The devs will have the ability to consume the provided components without being able to change the common configuration.
+- The users will be able to monitor their pipelines through the `internal_metrics` source.
+
+## Motivation
+
+- Helps Vector to grow organically within an organization by allowing teams to adopt Vector at their own pace without heavy administrator involvement.
+- Reduces the management overhead of devops/SREs by enabling teams to manage their own pipelines (spread the management load).
+
+## Implementation
+
+### How and where pipelines are stored
+
+To avoid backward incompatibility, pipelines will be loaded from a `pipelines` sub-directory relative to the Vector configuration directory (e.g., `/etc/vector/pipelines`). Therefore, if a user changes the location of the Vector configuration directory they will also change the `pipelines` directory path. They are coupled.
+
+The `pipelines` directory will contain all pipelines represented as individual files. For simplicity, and to ensure users do not overcomplicate pipeline management, sub-directories/nesting are not allowed. This is inspired by Terraform's single-level directory nesting, which has been a net positive for simple management of large Terraform projects.
+
+Each pipeline file is a processing subset of a larger Vector configuration file. Therefore, it follows the same syntax as Vector's configuration (`toml`, `yaml`, and `json`).
+
+### What is a pipeline
+
+- A pipeline is a collection of transforms that only have access to the component declared in the root configuration and their own pipeline's transforms.
+- A pipeline has an `id` being the name of the file without the extension. The pipeline `load-balancer.yml` will have `load-balancer` as its `id`.
+- Pipelines have access to any components from the root configuration. If the transform `foo` is defined in `/etc/vector/bar.toml`, it will be accessible by the pipeline `/etc/vector/pipelines/pipeline.toml`.
```

Comment:
```suggestion
- Pipelines have access to any components defined in the root configuration directory. For example, if the transform `foo` is defined in `/etc/vector/bar.toml`, it will be accessible by the pipeline `/etc/vector/pipelines/pipeline.toml`, but if a transform `bar` is defined in `/etc/vector/pipelines/another-pipeline.toml`, it will _not_ be accessible by other pipelines.
```